website/docs/contributing/ADRs/overarching/separation-request-response-schemas.md

```diff
@@ -0,0 +1,41 @@
+
+## Decision
+
+After careful consideration and discussion, it has been decided to separate the request and response schemas to address the challenges we have encountered. By creating distinct schemas for requests and responses, we aim to improve the flexibility and precision of our API documentation.
+
+### Advantages
+
+Separating the schemas will allow us to establish more precise and constrained response types while enabling more forgiving request types. This approach will facilitate better alignment between the expected data structures and the actual data transmitted in API interactions.
+
+The separation of request and response schemas will provide the following benefits:
+
+1. **Enhanced clarity and correctness**: With dedicated schemas for requests and responses, we can define the precise structure and constraints for each interaction. This will help prevent situations where the schemas are overly permissive or restrictive, improving the accuracy and clarity of the API documentation.
+
+2. **Improved maintainability**: By avoiding the reuse of schemas between requests and responses, we can modify and update them independently. This decoupling of schemas will simplify maintenance efforts and minimize the risk of unintended side effects caused by changes in one context affecting the other.
+
+3. **Flexibility for future enhancements**: Separating request and response schemas lays the foundation for introducing additional validation or transformation logic specific to each type of interaction. This modularity will enable us to incorporate future enhancements, such as custom validation rules or middleware, with ease.
+
+### Concerns
+
+While this decision brings several benefits, we acknowledge the following concerns that may arise from the separation of request and response schemas:
+
+1. **Increased schema maintenance**: By having separate schemas for requests and responses, there will be a need to maintain and update two sets of schemas instead of a single shared schema. This could potentially increase the maintenance overhead and introduce the possibility of inconsistencies between the two schemas.
+
+2. **Data duplication and redundancy**: With the separation of schemas, there might be instances where certain data fields or structures are duplicated between the request and response schemas. This redundancy could lead to code duplication and increase the risk of inconsistencies if changes are not carefully synchronized between the two schemas.
+
+3. **Complexity of migration**: Migrating existing API implementations to adopt the separated schemas might introduce complexities and challenges. Ensuring a smooth transition without disrupting ongoing development or existing integrations requires careful planning and coordination.
```

Comment:
If I understand correctly, the plan is here is do liberally apply the boyscout rule here. If that's the case then I think it's unlikely we'll do a complex migration. Only that new changes are required to incorporate this decision

Reply:
True, I do not believe a big bang migration replacing all schemas at once is feasible. But applying the boyscout rule (leave things better than you found it) would mean eventually all code that is in use will be updated to separate the schemas.

We should still aim to complete the migration by the release of 6.0

Reply:
Good point. I yeeted this concern and added some details about our migration strategy on the conclusion:
 - https://github.com/Unleash/unleash/pull/3869/commits/81809e7b09eee11a3bd2dcaed10fa54941d3e537
 - https://github.com/Unleash/unleash/pull/3869/commits/70148a3a3ea2d6804f82e1ee5a13d8125ea5299d